### ISSUE TITLE
tls_sender crash when doing key_update

### ISSUE THREAD

**Issue by zzydxm** [erlang/otp#6519]
**Describe the bug**
tls_sender will add a {key_update, From} event to itself when sequence number exceed
https://github.com/erlang/otp/blob/master/lib/ssl/src/tls_sender.erl#L478
But it does not have logic to handle this event and crashes.
```
tls_sender.erl:444 == error: REPORT_CB/1 CRASH: #{event => "TLS sender received unexpected event",reason => [{type,internal},{message,{key_update,{<0.11448.4>,#Ref<0.4237742432.3413377041.205359>}}}]}; Reason: {error,function_clause,[{ssl_logger,format,[#{event => "TLS sender received unexpected event",reason => [{type,internal},{message,{key_update,{<0.11448.4>,#Ref<0.4237742432.3413377041.205359>}}}]}],[{file,"ssl_logger.erl"},{line,81}]}]}
```

**To Reproduce**
Send more than 388736063997 packets in a ssl connection with cipher #{cipher => aes_256_gcm, key_exchange => any, mac => aead, prf => sha384}

**Expected behavior**
key_update event should be handled properly

**Affected versions**
25.0.2

**Comment by potatosalad** [comment]
It almost looks like https://github.com/erlang/otp/commit/ffa98227d95a09cd5aa7f243a4e61088b8283b1d was (potentially) merged incorrectly back in 2020:

1. Feb 4, 2020 - adds support for {next_event, internal, {key_update, ...}} (see also connection(internal, {key_update, ...}, ...) handler) https://github.com/erlang/otp/commit/80a06be340b3dd8df720d913866cc91b7eb489c5
2. Feb 11, 2020 - removes both the handler and the internal event for key_update https://github.com/erlang/otp/commit/e3c429adf59f0ef4be5ea4393265a096f8d0902b
3. Feb 12, 2020 - merges the above two commits, but includes the internal event without a corresponding handler https://github.com/erlang/otp/commit/ffa98227d95a09cd5aa7f243a4e61088b8283b1d

**Comment by u3s** [comment]
fix merged and planned for 25.2
